tests/json_predicate/binary_predicate_test.py

```diff
@@ -195,6 +195,168 @@ def test_dict_ne(self):
     self.assertGoodResult(PathValue('', {'a': 'A'}),
                           ne_pred, ne_pred(context, {'a': 'A'}))
 
+  def test_dict_match_simple_ok(self):
+    context = ExecutionContext()
+    source = {'n' : 10}
+    want = {'n' : jp.NUM_LE(20)}
+    result = jp.DICT_MATCHES(want)(context, source)
+
+    expect = (jp.CompositePredicateResultBuilder(jp.DICT_MATCHES(want))
+              .append_result(
+                  jp.PathPredicateResultBuilder(
+                      source=source,
+                      pred=jp.NUM_LE(20))
+                  .add_result_candidate(
+                      jp.PathValue('n', 10), jp.NUM_LE(20)(context, 10))
+                  .build(True))
+              .build(True))
+
+    self.assertTrue(result)
+    self.assertEquals(expect, result)
+
+  def test_dict_match_simple_bad(self):
+    context = ExecutionContext()
+    source = {'n' : 10}
+    want = {'n' : jp.NUM_NE(10)}
+    result = jp.DICT_MATCHES(want)(context, source)
+
+    expect = (jp.CompositePredicateResultBuilder(jp.DICT_MATCHES(want))
+              .append_result(
+                  jp.PathPredicateResultBuilder(
+                      source=source,
+                      pred=jp.NUM_NE(10))
+                  .add_result_candidate(
+                      jp.PathValue('n', 10), jp.NUM_NE(10)(context, 10))
+                  .build(False))
+              .build(False))
+
+    self.assertFalse(result)
+    self.assertEquals(expect, result)
+
+  def test_dict_match_multi_ok(self):
+    context = ExecutionContext()
+    source = {'a' : 'testing', 'n' : 10}
```

Comment:
In a normal integration test scenario, would `source` be the result coming back from something like a `list_resource`? How would that look connected into `DICT_MATCHES`.